tests/tests/vertex_indices/mod.rs

```diff
@@ -356,20 +337,17 @@ async fn vertex_index_common(ctx: TestingContext) {
         )
         .create_view(&wgpu::TextureViewDescriptor::default());
 
-    let mut tests = Vec::with_capacity(5 * 2 * 2 * 2);
+    let mut tests = Vec::with_capacity(5 * 2 * 2);
```

Comment:
thought(non-blocking): I understand why we're diminishing the capacity here, but I think it's in the name of something that's not worth it.

issue(blocking): If the intent is to only allocate once for `tests`' storage (which is the only reason I can imagine we'd want to call `Vec::with_capacity` here), then this expression was wrong before, and is still wrong after. The old expression is correct _after_ removing `vertex_pulling_transforms`, i.e., this should have been `5 * 2 * 2 * 2 * 2` (case, ID source, draw call kind, encoder kind, _and_ vertex-pulling transform boolean) prior to the change.

suggestion(blocking): Let's just remove this hunk for now, _or_ (even better) add a commit fixing this mistake (i.e., adding `* 2`) before this change, and then immediately reverting it in the “intended” change's commit.

☝🏻CC:

* @cwfitzgerald as original author in <https://github.com/gfx-rs/wgpu/pull/4723>.
* @teoxoy as reviewer of <https://github.com/gfx-rs/wgpu/pull/4723> and the last changes made to this line at <https://github.com/gfx-rs/wgpu/pull/5254>, too.

Reply:
suggestion(non-blocking): I doubt we're affecting noticeably performance here, and I think in testing code like this, clarity trumps minor performance gains. Let's do one of two things:

1. Use `Itertools::cartesian_product` and `Iterator::collect` over these `ARRAY` constants (or something else that I'll mention in another comment about `strum`) to get what we want with fewer footguns and mental cache pressure:

    ```rust
    let tests = TestCase::ARRAY
        .cartesian_product(IdSource::ARRAY)
        .cartesian_product(DrawCallKind::ARRAY)
        .cartesian_product(EncoderKind::ARRAY)
        .map(|(((case, id_source), draw_call_kind), encoder_kind)| Test {
            case,
            id_source,
            draw_call_kind,
            encoder_kind,
        })
        .collect::<Vec<_>>();
    ```
2. Just replace `tests`' initial value with a `Vec::new`, and not worry about re-allocation with such few elements:

    ```rust
    let mut tests = Vec::new();
    ```


☝🏻CC folks who participated in introducing this code in <https://github.com/gfx-rs/wgpu/pull/4723> (@cwfitzgerald as author, @teoxoy as reviewer).

Reply:
Resolved the blocking feedback I've raised in <https://github.com/gfx-rs/wgpu/pull/5773#discussion_r1645009650> with 53ddbc92 and 86fff753, and a new version of the original revision I've reviewed (895c0b42).

No further blocking feedback here; will file a follow-up for the remaining non-blocking stuff momentarily.

Reply:
Let's move non-blocking feedback in this convo to <https://github.com/gfx-rs/wgpu/pull/5840>. Resolving.